packages/cli/src/options/beaconNodeOptions/execution.ts

```diff
@@ -72,9 +75,8 @@ export const options: CliCommandOptions<ExecutionEngineArgs> = {
   },
 
   "execution.engineMock": {
-    description: "Set the execution engine to mock mode",
+    description: "Set the execution engine to mock mode (development only)",
```

Comment:
> Can we log a warn / error if we detect an attached validator + engineMock ?

Not sure where to hook in for this, what's the worst thing that can happen?

We can keep it hidden as well, but will likely cause people asking for something like this again.

Reply:
> Not sure where to hook in for this, what's the worst thing that can happen?

It's part of the specs, the bellatrix honest validator guide [here](https://github.com/ethereum/consensus-specs/blob/dev/specs/bellatrix/validator.md#beacon-chain-responsibilities):
> Note: A validator must not propose on or attest to a block that isn't deemed valid, i.e. hasn't yet passed the beacon chain state transition and execution validations.

A malicious proposer can perform arbitrary EL state transition or data withholding attack, and if lots of the network runs with this, then the chain splits.

Reply:
> A malicious proposer can perform arbitrary EL state transition or data withholding attack, and if lots of the network runs with this, then the chain splits.

That's clear to me as we vote for any EL payload without validating it, but seems bit far fetched and for an individual operator enabling this for whatever reason is not risk as far as I can see.

The engine mock in whatever shape (previous or in this PR) does not follow the spec. Need to see if we can have some way to error or at minimum emit a warning if used with attached validators as you suggested. Might be for the best with or without the changes in this PR.

Otherwise, we can just hide the flag.

Reply:
I think we should hide the flag anyway, letting users to use this feature seems dangerous, it's supposed for dev or testing purpose only
also we should not allow for mainnet and node has connected validators

Reply:
> also we should not allow for mainnet and node has connected validators

what about we keep it unhidden but have a hard coded check to fail startup if `--network=mainnet`?

Reply:
> We can keep it hidden as well, but will likely cause people asking for something like this again.

Was there a lot of people asking for this? If yes, would rather have a section in the doc to mention this with a big warning instead of unhiding the flag

Reply:
> Was there a lot of people asking for this?

Not a lot, but it's even nice for us to test sync or other issues on any network. I also linked a discussion, they are currently maintaining a fork of Lodestar with (similar) changes in this PR.